File: ArtFragment/app/src/main/java/com/alierdemalkoc/artfragment/roomdb/ArtDao.java
package com.alierdemalkoc.artfragment.roomdb;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.alierdemalkoc.artfragment.model.ArtModel;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Flowable;

@Dao
public interface ArtDao {
    @Query("SELECT name, id FROM Arts")
    Flowable<List<ArtModel>> getArtWithNameAndId();

    @Query("SELECT * FROM Arts WHERE id = :id")
    Flowable<ArtModel> getArtById(int id);

    @Insert
    Completable insert(ArtModel artModel);

    @Delete
    Completable delete(ArtModel artModel);


}

File: ArtBookTest/app/src/main/java/com/alierdemalkoc/artbooktest/roomdb/ArtDao.kt
package com.alierdemalkoc.artbooktest.roomdb

import androidx.lifecycle.LiveData
import androidx.room.*
import com.alierdemalkoc.artbooktest.model.Art

@Dao
interface ArtDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertArt(art: Art)

    @Delete
    suspend fun deleteArt(art: Art)

    @Query("SELECT * FROM arts")
    fun observeArts(): LiveData<List<Art>>
}